### PR TITLE
Upload New Supporting Document

### PR DESCRIPTION
**Pull request by ghickman**
This adds the PacketFile model and view/form for uploading a document.

Ref #1331

### CODE REVIEW

File: opentech/apply/projects/views.py [hunk 1]
```diff
@@ -106,25 +106,54 @@ def form_valid(self, form):
...
         RejectionView,
         SendForApprovalView,
         UpdateLeadView,
+        UploadDocumentView,
     ]
     model = Project
     template_name_suffix = '_admin_detail'
 
     def get_context_data(self, **kwargs):
+        categories = (DocumentCategory.objects.exclude(packet_files__project=self.object)
```
Comment (todd-dembrey) [inline opentech/apply/projects/views.py]: This wont cut it. It ignores the minimum required aspect of the data.

We should choose one of the following:
1. iterate over the relations in a dict to calculate what is left to add 
2. do nothing and list all required

1. is probably the most value in this instance i think?

There is probably also a query we could do to calculate it, but that sounds ott.

File: opentech/apply/projects/models.py [hunk 2]
```diff
@@ -22,6 +22,17 @@ def __str__(self):
         return f'Approval of "{self.project.title}" by {self.by}'
 
 
+class PacketFile(models.Model):
+    category = models.ForeignKey("DocumentCategory", on_delete=models.CASCADE, related_name="packet_files")
```
Comment (todd-dembrey) [inline opentech/apply/projects/models.py]: `nullable`

File: opentech/apply/activity/messaging.py [hunk 3]
```diff
@@ -742,6 +742,7 @@ class DjangoMessagesAdapter(AdapterBase):
         MESSAGES.BATCH_REVIEWERS_UPDATED: 'batch_reviewers_updated',
         MESSAGES.BATCH_TRANSITION: 'batch_transition',
         MESSAGES.BATCH_DETERMINATION_OUTCOME: 'batch_determinations',
+        MESSAGES.UPLOAD_DOCUMENT: 'Successfully uploaded document "{title}"',
```
Comment (todd-dembrey) [inline opentech/apply/activity/messaging.py]: :+1:

File: opentech/apply/projects/models.py (hunk 2 above)
Comment (ghickman) [reply opentech/apply/projects/models.py]: Fixed and rebased.

File: opentech/apply/projects/views.py (hunk 1 above)
Comment (ghickman) [reply opentech/apply/projects/views.py]: Fixed in 6497d047

File: opentech/apply/projects/views.py [hunk 4]
```diff
@@ -106,25 +106,54 @@ def form_valid(self, form):
...
         context['approvals'] = self.object.approvals.distinct('by')
-        context['remaining_document_categories'] = DocumentCategory.objects.all()
+        context['remaining_document_categories'] = categories
         return context
 
 
 class ApplicantProjectDetailView(DetailView):
+    form_views = [
+        UploadDocumentView,
+    ]
```
[the remark this answers is not in the document]
Comment (ghickman) [reply opentech/apply/projects/views.py]: Fixed and rebased.

File: opentech/apply/projects/models.py [hunk 5]
```diff
@@ -22,6 +22,17 @@ def __str__(self):
...
+    project = models.ForeignKey("Project", on_delete=models.CASCADE, related_name="packet_files")
+
+    title = models.TextField()
+    document = models.FileField(upload_to='supporting_documents/')
```
[the remark this answers is not in the document]
Comment (ghickman) [reply opentech/apply/projects/models.py]: Fixed and rebased.

**Pull request merged**